tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_add.rs

```diff
@@ -69,24 +90,67 @@ where
     default_add_test(param, executor);
 }
 
-fn integer_advanced_add_assign_with_carry_at_least_4_bits<P>(param: P)
+fn integer_extensive_trivial_default_add<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    let executor = CpuFunctionExecutor::new(&ServerKey::add_parallelized);
+    extensive_trivial_default_add_test(param, executor);
+}
+
+fn integer_advanced_overflowing_add_assign_with_carry_at_least_4_bits<P>(param: P)
 where
     P: Into<PBSParameters>,
 {
     // We explicitly call the 4 bit function to make sure it's being tested,
     // no matter the number of blocks / threads available
     let func = |sks: &ServerKey, lhs: &RadixCiphertext, rhs: &RadixCiphertext| {
         let mut result = lhs.clone();
-        sks.advanced_add_assign_with_carry_at_least_4_bits(
-            &mut result.blocks,
-            &rhs.blocks,
-            None,
-            OutputFlag::None,
-        );
-        result
+        if !result.block_carries_are_empty() {
+            sks.full_propagate_parallelized(&mut result);
+        }
+        let mut tmp_rhs;
+        let rhs = if rhs.block_carries_are_empty() {
+            rhs
+        } else {
+            tmp_rhs = rhs.clone();
+            sks.full_propagate_parallelized(&mut tmp_rhs);
+            &tmp_rhs
+        };
+        let overflowed = sks
+            .advanced_add_assign_with_carry_at_least_4_bits(
+                &mut result.blocks,
+                &rhs.blocks,
+                None,
+                OutputFlag::Carry,
+            )
+            .unwrap();
+        (result, overflowed)
     };
     let executor = CpuFunctionExecutor::new(&func);
-    default_add_test(param, executor);
+    default_overflowing_add_test(param, executor);
+}
+
+fn integer_extensive_trivial_overflowing_advanced_add_assign_with_carry_at_least_4_bits<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    // We explicitly call the 4 bit function to make sure it's being tested,
+    // no matter the number of blocks / threads available
+    let func = |sks: &ServerKey, lhs: &RadixCiphertext, rhs: &RadixCiphertext| {
+        let mut result = lhs.clone();
+        let overflowed = sks
+            .advanced_add_assign_with_carry_at_least_4_bits(
+                &mut result.blocks,
+                &rhs.blocks,
+                None,
+                OutputFlag::Carry,
+            )
+            .unwrap();
+        (result, overflowed)
```

Comment:
no carry management here ? I see it done in other test functions

Reply:
Aded them just for future proofing if we later change the extensive_trivial test to also have non clean inputs